### PR TITLE
#72 task creation screen add picture to task and save it to db

### LINKED ISSUE #72: Task creation screen - Add picture to task and save it to db

### LINKED ISSUE #101: Fix build - remove gradle errors from log
Currently when building the app there are lots of gradle errors/warnings in build log (when building locally as well as on CI). Part of log:

Error:warning: Ignoring InnerClasses attribute for an anonymous inner class (freemarker.cache.FileTemplateLoader$1) that doesn't come with an associated EnclosingMethod attribute. This class was probably produced by a compiler that did not target the modern .class file format. The recommended solution is to recompile the class from source, using an up-to-date compiler and without specifying any "-target" type options. The consequence of ignoring this warning is that reflective operations on this class will incorrectly indicate that it is *not* an inner class.

The issue is to figure out a solution and apply it (probably update a build config)

### PR DESCRIPTION
**Pull request by mikolevy**
Lot of stuff. In general:
- Add select picture feature to task creation
- Implement necessary helper (approach to assets management)
- Implement proxy for file picker library
- Implement tests (activity, unit and sample robolectric) and rule for dagger mocking
- Some less important refactoring and improvements

I made some changes in comparison to approach to assets management document, if you find this changes proper please create an issue to update docs and assign it to me :)

Closes #72 
Closes #101

### CODE REVIEW

File: Friendly-plans/app/src/androidTest/java/pg/autyzm/friendly_plans/TaskCreateActivityTest.java [hunk 1]
```diff
@@ -87,22 +108,114 @@ public void When_AddingNewTask_Expect_NewTaskAddedToDB() throws InterruptedExcep
...
+        activityRule.getActivity().runOnUiThread(new Runnable() {
+            public void run() {
+                fragment.onActivityResult(FilePickerProxy.PICK_FILE_REQUEST,
+                        FilePickerActivity.RESULT_OK, data);
+            }
+        });
+        Thread.sleep(1000);
+    }
+
+    private void addSafeCopyToTestFiles() {
```
Comment (malsolec) [inline Friendly-plans/app/src/androidTest/java/pg/autyzm/friendly_plans/TaskCreateActivityTest.java]: What do you think about extracting method managing test assets to AssetRule, which would handle all these methods - configure expected test asset and clean it up after test ? https://carlosbecker.com/posts/junit-rules/

File: Friendly-plans/app/src/main/java/pg/autyzm/friendly_plans/App.java [hunk 2]
```diff
@@ -19,10 +20,15 @@ public void onCreate() {
...
                 .validationModule(new ValidationModule())
+                .filePickerModule(new FilePickerModule())
                 .build();
     }
 
     public AppComponent getAppComponent() {
         return appComponent;
     }
+
+    public void setComponent(AppComponent appComponent) {
```
Comment (malsolec) [inline Friendly-plans/app/src/main/java/pg/autyzm/friendly_plans/App.java]: Could it be other solution for mock tests then changing code? Could you try with reflection or power mockito to mock app component? Power mockito allows to mock private fields I think. What do you think? http://www.johnmullins.co/blog/2015/02/15/beginners-guide-to-using-mockito-and-powermockito-to-unit-test-java/

File: Friendly-plans/app/src/main/java/pg/autyzm/friendly_plans/asset/AssetType.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.commons.io.FilenameUtils;
+
+public final class AssetType {
+
+    public static final String PICTURE = "PICTURE";
+    public static final String SOUND = "SOUND";
```
Comment (mikolevy) [inline Friendly-plans/app/src/main/java/pg/autyzm/friendly_plans/asset/AssetType.java]: I think it's good idea. The only problem is that greenDao doesn't accept enum type but I'll implement helper toString method - it's better than validation + exception in dao

File: Friendly-plans/app/src/androidTest/java/pg/autyzm/friendly_plans/TaskCreateActivityTest.java (hunk 1 above)
Comment (mikolevy) [inline Friendly-plans/app/src/androidTest/java/pg/autyzm/friendly_plans/TaskCreateActivityTest.java]: It's great idea! After applying this changes test code looks much cleaner and assets' managing part is reusable :)

File: Friendly-plans/app/src/main/java/pg/autyzm/friendly_plans/App.java (hunk 2 above)
Comment (mikolevy) [inline Friendly-plans/app/src/main/java/pg/autyzm/friendly_plans/App.java]: Really nice :) There is an annotation on setComponent method in DaggerMock lib example (I just forget to add it) but this solution is much better.

**Pull request merged**